tests/routes/test_status.py

```diff
@@ -1,123 +1,82 @@
-from funcx_web_service.models.user import User
 from unittest.mock import call
-from unittest.mock import Mock
 
 from funcx_web_service.models.tasks import Task
+from funcx_web_service.models.user import User
 from tests.routes.app_test_base import AppTestBase
 
 
 class TestStatus(AppTestBase):
-    def test_get_status(self, mock_auth_client, mock_redis, mock_user: User, mocker):
-        get_rc = mocker.patch(
-            "funcx_web_service.routes.funcx.g_redis_client",
-            return_value=mock_redis)
-        rf_get = mocker.patch(
-            "funcx_web_service.models.tasks.RedisField.__get__",
-            return_value=True)
-
-        mocker.patch.object(Task, "user_id").__get__ = Mock(return_value=mock_user.id)
-        mock_expire = mocker.patch.object(Task, "_set_expire", return_value=123)
-        mock_exists = mocker.patch.object(Task, "exists", return_value=True)
-
-        task_id = "42"
-        task = Task(mock_redis(), task_id=task_id)
-        mock_from_id = mocker.patch.object(Task, "from_id", return_value=task)
-
-        result = self.client.get(f"/api/v1/tasks/{task_id}", headers={"Authorization": "my_token"})
-
-        mock_exists.assert_called()
-        mock_expire.assert_called()
-        mock_from_id.assert_called()
-        rf_get.assert_called()
-        get_rc.assert_called()
+    def test_get_status(
+        self, mock_auth_client, mock_redis_task_factory, mock_user: User
+    ):
+        mock_redis_task_factory("42")
+        result = self.client.get(
+            "/api/v1/tasks/42", headers={"Authorization": "my_token"}
+        )
```

Comment:
Looks like we missed the check on task expiration getting set.

Reply:
Do you mean the `mock_expire.assert_called()`? That shouldn't part of this test.
It was only in here because of how the mocking was done -- the factory fixture is now responsible for handling things like this.

Or is getting a task status supposed to update the expiration? In that case, I can update to get the outcome by checking task expiration separately.